error.go

```diff
@@ -92,3 +92,24 @@ func (e *RequestError) Error() string {
 func (e *RequestError) Unwrap() error {
 	return e.Err
 }
+
+func (e *ErrorResponse) UnmarshalJSON(data []byte) (err error) {
+	var rawMap map[string]json.RawMessage
+	err = json.Unmarshal(data, &rawMap)
+	if err != nil {
+		return
+	}
+	if _, ok := rawMap["error"]; !ok {
+		e.Error = &APIError{
+			Code: "unknown",
```

Comment:
@zjy282 This is not the code returned by Azure, so users might get confused.

Reply:
@vvatanabe Thank you for your suggestion. In the process of using Azure, I found a variety of error returns that do not conform to the OpenAI specification. I suggest adding the unknown error code type and putting it in the readme as a description to ensure that new unknown errors will be generated in future updates of Azure. Can let the user be treated as a separate logic, what do you think